java/component/keyple-plugin/remote/local/src/main/java/org/eclipse/keyple/plugin/remote/impl/LocalServiceClientImpl.java

```diff
@@ -61,25 +60,32 @@ private LocalServiceClientImpl(
 
   /**
    * (package-private)<br>
-   * Create an instance of this singleton service
+   * Create an instance of the service. If a service exists with the same name, the service is
+   * overridden.
    *
+   * @param serviceName identifier of the local service
    * @param withReaderObservation true if reader observation should be activated
    * @return a not null instance of the singleton
    */
   static LocalServiceClientImpl createInstance(
-      boolean withReaderObservation, ObservableReaderEventFilter eventFilter) {
-    uniqueInstance = new LocalServiceClientImpl(withReaderObservation, eventFilter);
-    return uniqueInstance;
+      String serviceName, boolean withReaderObservation, ObservableReaderEventFilter eventFilter) {
+    if (serviceInstances == null) {
+      serviceInstances = new HashMap<String, LocalServiceClientImpl>();
+    }
+    LocalServiceClientImpl localService =
+        new LocalServiceClientImpl(withReaderObservation, eventFilter);
+    serviceInstances.put(serviceName, localService);
+    return localService;
   }
 
   /**
    * (package-private)<br>
-   * Retrieve the instance of this singleton service
+   * Retrieve the service associated by its serviceName
    *
-   * @return a not null instance
+   * @return a nullable instance
```

Comment:
a not null instance.
Il faut ajouter un @throws IllegalStateException If there's no service having the provided name.